File: models/Trip.js
import mongoose  from "mongoose"

const TripSchema = new mongoose.Schema({
    userId: {type: mongoose.Schema.Types.ObjectId, ref: 'User'},
    vehicle: {type: mongoose.Schema.Types.ObjectId, ref: 'Vehicle'}, 
    startTime: { type: String, required: true }, 
    endTime: { type: String, required: true }, 
})

const Trip = mongoose.model('Trip', TripSchema)

export default Trip

File: app/src/main/java/com/example/test/models/Trip.kt
package com.example.test.models

import java.io.Serializable

data class Trip(
    val userId:String,
   val vehicle: Vehicle,
   val startTime: String,
   val endTime: String,
): Serializable {
   val _id: String?= null
   constructor(
      _id:String,
      userId:String,
      vehicle: Vehicle,
      startTime: String,
      endTime: String,
   ) : this(
       userId,
       vehicle,
       startTime,
       endTime,) }

